mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ChessBoard.kt

```diff
@@ -143,33 +163,41 @@ private class DraggingState(
  * @param state the [DraggingState] for this composable.
  * @param target the [Offset] at which the piece rests.
  * @param cellSize the size of each square / piece of the board.
+ * @param onClick the callback called when the piece is clicked.
  * @param onDrop the callback called when the piece is dropped.
  */
 private fun Modifier.movablePiece(
     state: DraggingState,
     target: Offset,
     cellSize: Float,
+    onClick: () -> Unit,
     onDrop: (Position) -> Unit,
 ): Modifier = composed {
   val currentTarget by rememberUpdatedState(target)
+  val currentOnClick by rememberUpdatedState(onClick)
   val currentOnDrop by rememberUpdatedState(onDrop)
   pointerInput(Unit) {
-    detectDragGestures(
-        onDragStart = {
-          state.offset = currentTarget
-          state.isDragging = true
-        },
-        onDrag = { change, dragAmount ->
-          change.consumeAllChanges()
-          state.offset += dragAmount
-        },
-        onDragEnd = {
-          state.isDragging = false
-          val (x, y) = state.offset / cellSize
-          currentOnDrop(Position(x.roundToInt(), y.roundToInt()))
-        },
-        onDragCancel = { state.isDragging = false },
-    )
+    coroutineScope {
+      launch { detectTapGestures { currentOnClick() } }
+      launch {
+        detectDragGestures(
+            onDragStart = {
+              state.offset = currentTarget
+              state.isDragging = true
+            },
+            onDrag = { change, dragAmount ->
+              change.consumeAllChanges()
+              state.offset += dragAmount
+            },
+            onDragEnd = {
+              state.isDragging = false
+              val (x, y) = state.offset / cellSize
```

Comment:
Ah it divides directly both ^^

Reply:
Yes. I discovered this when working on the first `ChessBoard` iteration, it's quite useful actually !